ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java

```diff
@@ -172,31 +174,54 @@ public ConsistencyCheckerTool(ClusterMap clusterMap, StoreKeyFactory storeKeyFac
    * @return {@code true} if no real inconsistent blobs. {@code false}
    * @throws Exception
    */
-  public boolean checkConsistency(File[] replicas) throws Exception {
-    Map<StoreKey, ReplicationStatus> blobIdToStatusMap = getBlobStatusByReplica(replicas);
-    return checkConsistency(blobIdToStatusMap, replicas.length).size() == 0;
+  public Pair<Boolean, Map<File, DumpIndexTool.IndexProcessingResults>> checkConsistency(File[] replicas)
+      throws Exception {
+    Pair<Boolean, Map<File, DumpIndexTool.IndexProcessingResults>> resultsByReplica =
+        getIndexProcessingResults(replicas);
+    boolean success = resultsByReplica.getFirst();
+    if (success) {
+      Map<StoreKey, ReplicationStatus> blobIdToStatusMap =
+          getBlobStatusByReplica(replicas, resultsByReplica.getSecond());
+      success = checkConsistency(blobIdToStatusMap, replicas.length).size() == 0;
+    }
+    return new Pair<>(success, resultsByReplica.getSecond());
+  }
+
+  /**
+   * Processes the indexes of each of the replicas and returns the results.
+   * @param replicas the replicas to process indexes for.
+   * @return a {@link Pair} whose first indicates whether all results were sane and whose second contains the map of
+   * individual results by replica.
+   * @throws Exception if there is any error in processing the indexes.
+   */
+  private Pair<Boolean, Map<File, DumpIndexTool.IndexProcessingResults>> getIndexProcessingResults(File[] replicas)
+      throws Exception {
+    long currentTimeMs = time.milliseconds();
+    Map<File, DumpIndexTool.IndexProcessingResults> results = new HashMap<>();
+    boolean sane = true;
+    for (File replica : replicas) {
+      logger.info("Processing segment files for replica {} ", replica);
+      DumpIndexTool.IndexProcessingResults result =
+          dumpIndexTool.processIndex(replica, filterSet, currentTimeMs, throttler);
+      sane = sane && result.isIndexSane();
+      results.put(replica, result);
```

Comment:
any value in continuing if a previous result is not sane?

Reply:
Yeah - just to get results for all of them in case the caller wants to look and see and act upon it if required. 

For example, the `DiskReformatter` skips sanity checks if the source itself is not sane. It sends both the src and the newly created store files as "replicas" - it needs results for both.